include/IECoreVDB/VDBObject.h

```diff
@@ -91,7 +91,7 @@ class IECOREVDB_API VDBObject : public IECoreScene::VisibleRenderable
 		//! empty for procedurally generated VDBs
 		std::string fileName() const
 		{
-			return m_fileName;
+			return m_file->filename();
```

Comment:
`m_file` can be null, so we should guard against that and return the empty string. I'm not sure we have test coverage for this, so it'd be good to add it.